functions/index.js

```diff
@@ -4,25 +4,31 @@ admin.initializeApp(functions.config().firebase);
 const db = admin.database();
 
 exports.addPost = functions.database
-  .ref("/post")
+  .ref("/post/{postId}")
   .onCreate((snapshot, context) => {
     db.ref("/amountPost")
       .once("value")
       .then(_snapshot =>
-        db.ref("/amountPost").set(_snapshot.val() ? _snapshot.val() + 1 : 1)
+        db
+          .ref("/amountPost")
+          .set(typeof _snapshot.val() === "number" ? _snapshot.val() + 1 : 1)
       )
       .catch(e => db.ref("/amountPost").set(1));
   });
 
 exports.deletePost = functions.database
-  .ref("/post")
+  .ref("/post/{postId}")
   .onDelete((snapshot, context) => {
     db.ref("/amountPost")
       .once("value")
       .then(_snapshot =>
         db
           .ref("/amountPost")
-          .set(_snapshot.val() && _snapshot.val() > 0 ? _snapshot.val() - 1 : 0)
+          .set(
+            typeof _snapshot.val() === "number" && _snapshot.val() > 0
+              ? _snapshot.val() - 1
+              : 0
+          )
       )
       .catch(e => db.ref("/amountPost").set(0));
```

Comment:
catch should probably not reset. an error could mean something not just number errors.